### ISSUE TITLE
Validation split with Dataloader/Dataset

### ISSUE THREAD

**Issue by sudarshan85** [pytorch/pytorch#1106]
Hi,

I've been searching for a way to do this but didn't come across any solution in the docs/tutorials/forums. I'm not sure whether its a feature and I don't know how to use it or whether its not implemented at all (if not just wanted to throw it out here to be considered). I'm new here and I'm working with the CIFAR10 dataset to start and get familiar with the pytorch framework. 

In the [tutorials](http://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html#sphx-glr-beginner-blitz-cifar10-tutorial-py), the data set is loaded and split into the trainset and test by using the train flag in the arguments. This is nice, but it doesn't give a validation set to work with for hyperparameter tuning.

Was this intentional or is there anyway to do this with dataloader? In particular, I was wonder whether the sequence of: (1) train mini-batches for n iterations keep track of acc and loss, (2) test on validation set, record acc and loss, (3) sample new hyperparameters; rinse and repeat (similar to what I've done in the cs231n Stanford course). 

Is there a way to do this in the current framework? I can imagine doing this I have manually loaded the data  (maybe using numpy) and have my (X_train, y_train, X_dev, y_dev, X_test, y_test) tuple and proceed in the fashion I've stated above, but I just wanted to check in here first.

Thanks.

**Comment by soumith** [comment]
the DataLoader gives no ability to split a dataset further. You will have to do that yourself. 
You can use a `SubsetRandomSampler` to achieve it, or a custom sampler of your choosing.

In the future, discussions such as these are more suited for https://discuss.pytorch.org/

**Comment by kevinzakka** [comment]
I've created an example [script](https://gist.github.com/kevinzakka/d33bf8d6c7f06a9d8c76d97a7879f5cb) which lets you create train, validation and test splits on torchvision datasets (i.e. from torchvision.datasets) with support for shuffling, transformations and optional grid plotting of the data. It should be applicable to any other dataset in that library with some minor tweaking.

**Comment by Fuchai** [comment]
Try [this snippet of mine](https://gist.github.com/Fuchai/12f2321e6c8fa53058f5eb23aeddb6ab) if you need to split Dataset into training and validation.
